components/map/basic/src/index.js

```diff
@@ -0,0 +1,227 @@
+    this.currentLibrariesLoaded = Array.from(document.querySelectorAll('script')).map((library) => library.src)
+  }
+
+  _buildMapUIComponents () {
+    this.map.ui = window.H.ui.UI.createDefault(this.map.instance, this.map.defaultLayers)
+  }
+
+  _makeInteractive () {
+    this.map.behavior = new window.H.mapevents.Behavior(new window.H.mapevents.MapEvents(this.map.instance))
+  }
+
+  _initStylesInjection (styles) {
+    const stylesLength = styles.length
+    const styleUrl = styles.shift()
+
+    if (!this.currentStylesLoaded.includes(styleUrl) && stylesLength) {
+      const link = document.createElement('link')
+      link.href = styleUrl
+      link.rel = 'stylesheet'
+      link.type = 'text/css'
+      document.body.appendChild(link)
+      this._initStylesInjection(styles)
+    } else if (stylesLength) {
+      this._initStylesInjection(styles)
+    }
+  }
+
+  _initLibraryInjection (libraries, appId, appCode, useCIT, useHTTPS) {
+    const libraryLength = libraries.length
+    const libraryUrl = libraries.shift()
+
+    if (!this.currentLibrariesLoaded.includes(libraryUrl) && libraryLength) {
+      const script = document.createElement('script')
+      script.src = libraryUrl
+      script.onload = () => this._initLibraryInjection(libraries, appId, appCode, useCIT, useHTTPS)
+      document.body.appendChild(script)
+    } else if (!libraryLength) {
+      this._initHerePlatform(appId, appCode, useCIT, useHTTPS)
+    } else {
+      this._initLibraryInjection(libraries, appId, appCode, useCIT, useHTTPS)
+    }
+  }
+
+  _initHerePlatform (appId, appCode, useCIT, useHTTPS) {
+    this.map.platform = new window.H.service.Platform({
+      useCIT,
+      useHTTPS,
+      app_id: appId,
+      app_code: appCode,
+
+    })
+    this.map.defaultLayers = this.map.platform.createDefaultLayers()
+    this._renderHereMap()
+  }
+
+  _renderHereMap () {
+    const map = this._buildMapInstance()
+    if (map) {
+      this.props.isInteractive && this._makeInteractive()
+      this.props.isInteractive && this._buildMapUIComponents()
+      this.props.marker && this._buildMarkers()
+      this.props.circleShape && this._buildCircleShape()
+    }
+  }
+
+  _buildMarkers () {
+    const generalIcon = new window.H.map.DomIcon(this.props.marker.genericIcon)
+    this.props.marker.elements.forEach(element => {
+      const icon = element.icon ? new window.H.map.DomIcon(element.icon) : generalIcon
+      this.map.instance.addObject(new window.H.map.DomMarker({ lat: element.lat, lng: element.lng }, { icon: icon }))
+    })
+  }
+
+  _buildCircleShape () {
+    this.props.circleShape.elements.forEach(element => {
+      this.map.instance.addObject(
+        new window.H.map.Circle({lat: element.lat, lng: element.lng},
+        element.radius,
+        { style: element.style || this.props.circleShape.styleGeneric })
+      )
+    })
+  }
+
+  _buildMapInstance () {
+    const DOMElementToBeFilled = document.getElementById('sui-here-map')
```

Comment:
If you use the ''sui-here-map'' literal more than once, you should store it as a contant.
Plus, you are using an id to reference your element. This means that you component is "singleton". Only one instance of it would work on a page, not two.

Use unique ids or react ref prop for that